### PR TITLE
Restart annotations with resolved filter; improve coverage

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/editor/SonarExternalAnnotator.java [hunk 1]
```diff
@@ -63,13 +63,15 @@ public void apply(@NotNull PsiFile file, AnnotationContext annotationResult, @No
...
-    for (LiveIssue issue : issues) {
-      // reject non-null ranges that are no longer valid. It probably means that they were deleted from the file.
-      RangeMarker range = issue.getRange();
-      if (range == null || range.isValid()) {
-        addAnnotation(issue, holder);
-      }
-    }
+    issues.stream()
+      .filter(issue -> !issue.isResolved())
+      .forEach((issue) -> {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/editor/SonarExternalAnnotator.java]: MINOR Remove the parentheses around the "issue" parameter [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1611)

File: src/main/java/org/sonarlint/intellij/core/ServerIssueUpdater.java [hunk 2]
```diff
@@ -140,4 +112,51 @@ public void disposeComponent() {
...
+        Iterator<ServerIssue> serverIssues = fetchServerIssues(server, engine, moduleKey, relativePath);
+        Collection<Trackable> serverIssuesTrackable = toStream(serverIssues).map(ServerIssueTrackable::new).collect(Collectors.toList());
+
+        if (!serverIssuesTrackable.isEmpty()) {
+          issueManager.matchWithServerIssues(virtualFile, serverIssuesTrackable);
+        }
+      } catch (Throwable t) {
+        // note: without catching Throwable, any exceptions raised in the thread will not be visible
+        console.error("error while fetching and matching server issues", t);
+        t.printStackTrace();
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/core/ServerIssueUpdater.java]: MINOR Use a logger to log this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1148)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/org/sonarlint/intellij/core/ServerIssueUpdaterTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  public void testServerIssueTracking() {
+    VirtualFile file = mock(VirtualFile.class);
+    ServerIssue serverIssue = mock(ServerIssue.class);
+    when(file.getPath()).thenReturn("");
+
+    // mock creation of engine / server
+    ConnectedSonarLintEngine engine = mock(ConnectedSonarLintEngine.class);
+    when(bindingManager.getConnectedEngine()).thenReturn(engine);
+    SonarQubeServer server = mock(SonarQubeServer.class);
+    when(server.getHostUrl()).thenReturn("http://localhost:9000");
```
Comment (janos-ss) [inline src/test/java/org/sonarlint/intellij/core/ServerIssueUpdaterTest.java]: maybe "dummyserver" is less misleading

**Pull request merged**